site/shared.templ

```diff
@@ -101,6 +101,90 @@ templ headerTopLevelLinks(r *http.Request) {
 	@headerTopLevelLink(r, "Bundler")
 }
 
+type SiteSearchSignals struct {
+	Search string `json:"siteSearch"`
+	SearchFetching bool `json:"searchFetching"`
+	OpenSearchResults bool `json:"openSearchResults"`
+}
+
+type SearchResult struct {
+	ID    string  `json:"id"`
+	Title string `json:"title"`
+	Score float64 `json:"score"`
+	Fragment string `json:"fragment"`
+}
+
+const SearchFromHeaderKey = "X-Search-From-Header"
+
+templ SiteSearch(signals *SiteSearchSignals, searchResults []SearchResult, fromHeader bool) {
+	<div data-signals={templ.JSONString(signals)} class="w-full">
+		<div class="flex flex-nowrap w-full">
+			<input 
+				id="site_search_input" 
+				class="flex-1 input input-bordered h-8 w-full" 
+				type="search"
+				placeholder="Search..."
+				data-bind="siteSearch"
+				data-on-input__debounce.300ms={ fmt.Sprintf(
+    				"$siteSearch.length > 1 && @get('/api/search', %s)", 
+   					 fmt.Sprintf(`{"headers": {"%s": "%t"}}`, SearchFromHeaderKey, fromHeader),
+				)}
+				data-indicator="$searchFetching"
+			/>
+			@sseIndicator("searchFetching")
+		</div>
+			<ul 
+				id="search_results" 
+				class={
+					"search-results", 
+					"w-[90vw]",
+					"lg:w-full",
+					"relative",
+					"lg:absolute",
+					templ.KV("lg:-left-full", len(searchResults) > 0) 
+				} 
+				data-on-click__outside="$openSearchResults ? $openSearchResults = false : null"
+				data-on-keydown__window="evt.key == 'Escape' && $openSearchResults ? $openSearchResults = false : null" 
+				data-show="$openSearchResults">
+				if len(searchResults) > 0 {
+					for _, result := range searchResults {
+						<li>
+							<a href={templ.SafeURL(result.ID)}>
+								<h3 class="border-b-1">{result.Title}</h3>
+								<div class="text-sm">
+									@templ.Raw(result.Fragment)
+								</div>
+							</a>
+						</li>
+					}
+				} else {
+					<p>No results found.</p>
+				}
+			</ul>
+	</div>
+}
+
+templ HeaderSiteSearch(signals *SiteSearchSignals, searchResults []SearchResult) {
+	<div id="header_site_search" class="hidden lg:flex lg:gap-4 relative z-10">
+		@SiteSearch(&SiteSearchSignals{
+			Search: signals.Search,
+			SearchFetching: signals.SearchFetching,
+			OpenSearchResults: signals.OpenSearchResults,
+		}, searchResults, true)
+	</div>
+}
+
+templ DrawerSiteSearch(signals *SiteSearchSignals, searchResults []SearchResult) {
```

Comment:
Is `DrawerSiteSearch` being used and, if so, could you explain what for?

Reply:
Ah, it appears to be be used for the mobile search. Is that correct?

Reply:
I’m guessing that the two `input` fields with `id="site_search_input"` is intentional too?

Reply:
Hey, yes the drawer site search is for the mobile input. And I've used a header value to provide a switch so it sends down the correct fragment. I'll take another look tomorrow as I'm out atm. It might be the case that the ID is not needed anymore when I refactored/added mobile.

Reply:
Just confirmed those ID's aren't needed anymore and removed them